hail/design-docs/tensors/pc_relate.py

```diff
@@ -0,0 +1,61 @@
+import hail as hl
+
+def gram(x):
+    return x @ x.T
+
+def pc_relate(mt, k):
+    # scores is samples-by-k matrix
+    _, scores, _ = hl.hwe_normalized_pca(mt.gt, k)
+
+    # mean impute missing data
+    mt = mt.select_entries(x = mt.gt.n_alt_alleles / 2)
+    mt = mt.annotate_rows(mean = hl.agg.mean(mt.x))
+    mt = mt.select_entries(x = hl.or_else(mt.x, mt.mean))
+    g = mt.to_ndarray(mt.x)
+
+    # X is k-by-variants matrix
+    X = hl.lstsq(scores, g.T)
```

Comment:
Is linear algebra going to be in its own sub-module?  I feel top-level is going to get crowded.

Reply:
I don't think we want to import our linear algebra stuff as `np`, since collected local matrices will be actual  numpy matrices and the users are going to want to still use numpy.